misc/vtk/src/processors/volumetovtk.cpp

```diff
@@ -164,14 +182,32 @@ void VolumeToVTK::process() {
                 throw Exception("Cannot map type to VTK.", IVW_CONTEXT_CUSTOM("VolumeToVtk"));
             }
         }
-    }();
+    };
 
     const auto nComp = volume->getDataFormat()->getComponents();
-    data_->AllocateScalars(vtkType, static_cast<int>(nComp));
+    data_->AllocateScalars(vtkType(volume->getDataFormat()->getId()), static_cast<int>(nComp));
     void* dst = data_->GetScalarPointer();
     const void* src = volume->getRepresentation<VolumeRAM>()->getData();
     std::memcpy(dst, src, glm::compMul(dim) * volume->getDataFormat()->getSizeInBytes());
 
+    volumeData_.clear();
+    for (auto&& [outport, v] : optionalVolumes_.getSourceVectorData()) {
+        auto* dataFormat = v->getDataFormat();
+
+        auto scalars = vtkSmartPointer<vtkDataArray>(
+            vtkDataArray::CreateDataArray(vtkType(dataFormat->getId())));
+        scalars->SetNumberOfComponents(static_cast<int>(dataFormat->getComponents()));
+        scalars->SetName(outport->getProcessor()->getIdentifier().c_str());
+        scalars->SetNumberOfTuples(glm::compMul(dim));
+
+        void* dstData = scalars->GetVoidPointer(0);
+        const void* srcData = v->getRepresentation<VolumeRAM>()->getData();
```

Comment:
Wonder if we could create a vtk data array that just points to the volume ram without creating the copy?

Reply:
https://vtk.org/doc/nightly/html/classvtkImplicitArray.html

Might work even if a bit odd